mozilla_aws_cli/sts_conn.py

```diff
@@ -43,20 +48,32 @@ def get_credentials(bearer_token, id_token_dict, role_arn):
 
             # Call the STS API
             resp = requests.get(url=sts_url, params=parameters)
+            try:
+                root = ElementTree.fromstring(resp.content)
+            except ElementTree.ParseError as e:
+                raise MalformedResponseWarning('Unable to parse XML response to AssumeRoleWithWebIdentity call')
             if resp.status_code != requests.codes.ok:
-                if 'The requested DurationSeconds exceeds the MaxSessionDuration set for this role' in resp.text:
+                error = dict(
+                    [(x.tag.split('}', 1)[-1], x.text) for x in root.find(
+                        './sts:Error',
+                        {'sts': 'https://sts.amazonaws.com/doc/2011-06-15/'})])
+                logger.error(
+                    'AWS STS Call failed {status} {Type} {Code} : {Message}'.format(
+                        status=resp.status_code, **error))
+                if (error['Code'] == 'ValidationError'
```

Comment:
Can we just key off of `Code == ValidationError`? I'm sure it's not a 1:1 mapping, but if it's the only thing in our code that triggers it then it seems fine?

Reply:
No, since there are other things that can cause a `ValidationError` besides this max duration issue. If we did, we could detect other, non-duration-related, validation errors and start our duration backoff which would be wrong.